src/custom/state/enhancedTransactions/hooks/TransactionHooksMod.tsx

```diff
@@ -0,0 +1,104 @@
+// import { TransactionResponse } from '@ethersproject/providers'
+import { /* useCallback,*/ useMemo } from 'react'
+import { /* useAppDispatch,*/ useAppSelector } from 'state/hooks'
+
+import { useActiveWeb3React } from 'hooks/web3'
+// import { addTransaction } from '../actions'
+import { EnhancedTransactionDetails } from '../reducer'
+
+// // helper that can take a ethers library transaction response and add it to the list of transactions
+// export function useTransactionAdder(): (
+//   response: TransactionResponse,
+//   customData?: { summary?: string; approval?: { tokenAddress: string; spender: string }; claim?: { recipient: string } }
+// ) => void {
+//   const { chainId, account } = useActiveWeb3React()
+//   const dispatch = useAppDispatch()
+
+//   return useCallback(
+//     (
+//       response: TransactionResponse,
+//       {
+//         summary,
+//         approval,
+//         claim,
+//       }: { summary?: string; claim?: { recipient: string }; approval?: { tokenAddress: string; spender: string } } = {}
+//     ) => {
+//       if (!account) return
+//       if (!chainId) return
+
+//       const { hash } = response
+//       if (!hash) {
+//         throw Error('No transaction hash found.')
+//       }
+//       dispatch(addTransaction({ hash, from: account, chainId, approval, summary, claim }))
+//     },
+//     [dispatch, chainId, account]
+//   )
+// }
+
+// returns all the transactions for the current chain
+export function useAllTransactions(): { [txHash: string]: EnhancedTransactionDetails } {
+  const { chainId } = useActiveWeb3React()
+
+  const state = useAppSelector((state) => state.transactions)
+
+  return chainId ? state[chainId] ?? {} : {}
+}
+
+export function useIsTransactionPending(transactionHash?: string): boolean {
+  const transactions = useAllTransactions()
+
+  if (!transactionHash || !transactions[transactionHash]) return false
+
+  return !transactions[transactionHash].receipt
+}
+
+/**
+ * Returns whether a transaction happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param tx to check for recency
+ */
+export function isTransactionRecent(tx: EnhancedTransactionDetails): boolean {
+  return new Date().getTime() - tx.addedTime < 86_400_000
```

Comment:
is this a mod? if so make the magic number go away pls

Reply:
Because is a mod, i would avoid to change it. But is a good point, however, is not part of this PR, plus I think Leandro touched near this in the orders PR, so I think is not worth it